six/modules/c++/six.sicd/source/ComplexData.cpp

```diff
@@ -56,38 +56,42 @@ void ComplexData::getOutputPlaneOffsetAndExtent(
         types::RowCol<size_t>& offset,
         types::RowCol<size_t>& extent) const
 {
-    offset.row = offset.col = 0;
-
     const six::sicd::AreaPlane& areaPlane =
         *radarCollection->area->plane;
+    getOutputPlaneOffsetAndExtent(offset, extent, areaPlane);
+}
+
+void ComplexData::getOutputPlaneOffsetAndExtent(
+        types::RowCol<size_t>& offset,
+        types::RowCol<size_t>& extent,
+        const AreaPlane& areaPlane) const
+{
+    offset.row = offset.col = 0;
     extent.row = areaPlane.xDirection->elements;
     extent.col = areaPlane.yDirection->elements;
 
     if (imageFormation.get())
     {
         const std::string& segmentID =
                 imageFormation->segmentIdentifier;
+        try
+        {
+            const six::sicd::Segment& segment = areaPlane.getSegment(segmentID);
```

Comment:
Yes, the logic you have above is right.

I think the logic you want here is to just replace the try/catch with a check to see if `segmentID` is empty.

I think we can avoid adding any new `hasSegment()` or `getSegmentIndex()` methods for now.  As a heads up, there will usually be a very small number of segments.   If it was 100, that would be insanely big.  So performance won't be an issue.  What I like about `getSegment()` is that the caller can be lazy with no error checking and act like it'll succeed (and in the rare case something is awry with the SICD or their logic, it will throw).